src/main/scala/org/apache/spark/streamdm/classifiers/trees/Node.scala

```diff
@@ -17,65 +17,70 @@
 
 package org.apache.spark.streamdm.classifiers.trees
 
-import scala.collection.mutable.ArrayBuffer
-import scala.math.max
+import java.util.Random
+
+import org.apache.spark.internal.Logging
 
+import scala.collection.mutable.{ArrayBuffer, ListBuffer}
+import scala.math.max
 import org.apache.spark.streamdm.core._
 import org.apache.spark.streamdm.core.specification._
 import org.apache.spark.streamdm.classifiers.bayes._
 import org.apache.spark.streamdm.utils.Utils.argmax
 
 /**
- * Abstract class containing the node information for the Hoeffding trees.
- */
-abstract class Node(val classDistribution: Array[Double]) extends Serializable {
+  * Abstract class containing the node information for the Hoeffding trees.
+  */
+abstract class Node(val classDistribution: Array[Double]) extends Serializable with Logging {
```

Comment:
I've spited it in one file per class, so it is now 8 files instead of just 1. To make it more organized I've created a package within the trees package named nodes.